packages/netsuite-adapter/src/autogen/types.ts

```diff
@@ -13,190 +13,432 @@
+import { usereventscriptType } from './types/custom_types/usereventscript'
+import { workbookType } from './types/custom_types/workbook'
+import { workbookbuilderpluginType } from './types/custom_types/workbookbuilderplugin'
+import { workflowType } from './types/custom_types/workflow'
+import { workflowactionscriptType } from './types/custom_types/workflowactionscript'
 
+const customTypesNamesList = [
+  'addressForm',
+  'advancedpdftemplate',
+  'bankstatementparserplugin',
+  'bundleinstallationscript',
+  'center',
+  'centercategory',
+  'centerlink',
+  'centertab',
+  'clientscript',
+  'cmscontenttype',
+  'crmcustomfield',
+  'customglplugin',
+  'customlist',
+  'customrecordactionscript',
+  'customrecordtype',
+  'customsegment',
+  'customtransactiontype',
+  'dataset',
+  'datasetbuilderplugin',
+  'emailcaptureplugin',
+  'emailtemplate',
+  'entitycustomfield',
+  'entryForm',
+  'ficonnectivityplugin',
+  'fiparserplugin',
+  'integration',
+  'itemcustomfield',
+  'itemnumbercustomfield',
+  'itemoptioncustomfield',
+  'kpiscorecard',
+  'mapreducescript',
+  'massupdatescript',
+  'othercustomfield',
+  'pluginimplementation',
+  'plugintype',
+  'portlet',
+  'promotionsplugin',
+  'publisheddashboard',
+  'restlet',
+  'role',
+  'savedcsvimport',
+  'savedsearch',
+  'scheduledscript',
+  'sdfinstallationscript',
+  'secret',
+  'sspapplication',
+  'sublist',
+  'subtab',
+  'suitelet',
+  'transactionForm',
+  'transactionbodycustomfield',
+  'transactioncolumncustomfield',
+  'translationcollection',
+  'usereventscript',
+  'workbook',
+  'workbookbuilderplugin',
+  'workflow',
+  'workflowactionscript',
+] as const
 
-export type CustomType = 'addressForm' | 'advancedpdftemplate' | 'bankstatementparserplugin' | 'bundleinstallationscript' | 'center' | 'centercategory' | 'centerlink' | 'centertab' | 'clientscript' | 'cmscontenttype' | 'crmcustomfield' | 'customglplugin' | 'customlist' | 'customrecordactionscript' | 'customrecordtype' | 'customsegment' | 'customtransactiontype' | 'dataset' | 'datasetbuilderplugin' | 'emailcaptureplugin' | 'emailtemplate' | 'entitycustomfield' | 'entryForm' | 'ficonnectivityplugin' | 'fiparserplugin' | 'integration' | 'itemcustomfield' | 'itemnumbercustomfield' | 'itemoptioncustomfield' | 'kpiscorecard' | 'mapreducescript' | 'massupdatescript' | 'othercustomfield' | 'pluginimplementation' | 'plugintype' | 'portlet' | 'promotionsplugin' | 'publisheddashboard' | 'restlet' | 'role' | 'savedcsvimport' | 'savedsearch' | 'scheduledscript' | 'sdfinstallationscript' | 'secret' | 'sspapplication' | 'sublist' | 'subtab' | 'suitelet' | 'transactionForm' | 'transactionbodycustomfield' | 'transactioncolumncustomfield' | 'translationcollection' | 'usereventscript' | 'workbook' | 'workbookbuilderplugin' | 'workflow' | 'workflowactionscript'
+const customTypesNamesSet: ReadonlySet<string> = new Set(customTypesNamesList)
```

Comment:
nit:
get be written a bit lower in the file and then use `CustomType`
```suggestion
const customTypesNamesSet: ReadonlySet<CustomType> = new Set(customTypesNamesList)
```

Reply:
I think that the external API is better to use `CustomType` and in `isCustomTypeName` use `customTypesNamesSet.has(name as CustomType)`.

I know that it uses a wrong casting here but it's a safe one and this way you are very clear with the type of `customTypesNamesSet` which makes it safer.

I found https://dev.to/hansott/how-to-check-if-string-is-member-of-union-type-1j4m which agrees with my intuition here, but again this is a nit so if you strongly objects, feel free to ignore :)